### ISSUE TITLE
ls panics on binary file

### ISSUE THREAD

**Issue by jhod0** [redox-os/coreutils#43]
It seems that ls panics on an out-of-range error when given a binary file:
```
# ls test.wav
assertion failed: index < self.len() in rust/src/libcore/slice.rs:510
```

Obviously, this isn't ls's intended use, but I assume it should detect test.wav is a binary file and give an informative error message.

**Comment by ids1024** [comment]
It looks like this was fixed in commit 0be6f083fcb218e2b4f7c74a487942ff7c5ad8cb.

**Comment by jackpot51** [comment]
yes,  it was. thanks
